### ISSUE TITLE
[Feature Proposal] Self-contained quick starter

### ISSUE THREAD

**Issue by maronuu** [m3dev/kannon#5]
## What

Self-contained quick starter for kannon library is essential for people to get started with gokart&kannon.

### Using Jupyter Notebook
Using jupyter notebook enables to avoid multiple `pyproject.toml` settings for tutorials & library, since you can `pip install kannon` within a notebook. Within the notebook, users install `docker`, `minikube`, `kannon` and their dependencies.

### Using [minikube](https://kubernetes.io/ja/docs/setup/learning-environment/minikube/)
Minikube is local Kubernetes, focusing on making it easy to learn and develop for Kubernetes.

It supports:
- [Deploying an app](https://minikube.sigs.k8s.io/docs/handbook/deploying/)
- [Pushing image to minikube](https://minikube.sigs.k8s.io/docs/handbook/pushing/)

Notebooks support bash commands and [a magic command to write an cell content to a file](https://ipython.readthedocs.io/en/stable/interactive/magics.html), so all steps can be included in a single notebook.

### Using PersistentVolume (PV) and PersistentVolumeClaim (PVC) as a common cache
Create PV and PVC in minikube cluster. And then, create a master job who uses the volume as a cache.

**Comment by maronuu** [comment]
In the current implementation, we cannot let child jobs use a mounted k8s PV as a cache since there is no option `volumes` and `volumeMounts` in [master.py](https://github.com/m3dev/kannon/blob/main/kannon/master.py#L141) :(

An ideal job yaml is like this:
```yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: job-quick-starter
spec:
  template:
    spec:
      containers:
      - name: job
        image: kannon_quick_starter
        command: ["python", "./run_batch.py", "--container-name", "quick-starter", "--image-name", "kannon_quick_starter"]
        imagePullPolicy: IfNotPresent
        volumeMounts:
        - name: cache-volume
          mountPath: /data/cache
        env:
          - name: TASK_WORKSPACE_DIRECTORY
            value: /data/cache
      restartPolicy: Never
      volumes:
      - name: cache-volume
        persistentVolumeClaim:
          claimName: cache-volume-claim
```

And child jobs also need to have `volumes` and `volumeMounts` values.

## Possible solutions

I believe method 1 is better than method 2.

### 1. `kannon.Kannon` arg
To allow to give them as arguments of `kannon.Kannon` object.
```python
Kannon(
        # new arguments!
        volume_mounts = [{"cache-volume", "./cache"}],
        volumes = [...]
    ).build(task_root)
```

### 2. `master.py` creates job objects from a given yaml files.
COPY k8s yamls to docker container. When `master.py` creates job objects, it can load the yamls in the container local.

**Comment by maronuu** [comment]
> Possible solutions

Method 1 can be implemented with a tiny change of codes. #6 is working on this.

**Comment by maronuu** [comment]
### Notebook may not be necessary

The reason why I first introduced jupyter notebook is that the quick starter will be dependent on `kannon` library, which will lead to complex `pyproject.toml`. `kannon` can be installed via pip in a notebook, so it will be solved.

However, `kannon` is required for a master job and child jobs, which run on containers with the same single image. So, installing `kannon` in `Dockerfile` is all we need.

The proposed quick starter idea is like [this](https://github.com/maronuu/kannon-example-local). A detailed explanation of the master/child.py scripts and clear documentation on the procedure of job execution.
I'm working on it now.

**Comment by yokomotod** [comment]
@maronuu Will you close this issue ? seems to be resolved by #17
